CMakeLists.txt

```diff
@@ -100,7 +102,12 @@ CHECK_INCLUDE_FILES("sys/types.h" HAVE_SYS_TYPES_H)
 CHECK_INCLUDE_FILES("sys/vfs.h" HAVE_SYS_VFS_H)
 CHECK_INCLUDE_FILES("sys/prctl.h" HAVE_SYS_PRCTL_H)
 CHECK_INCLUDE_FILES("execinfo.h" HAVE_EXECINFO_H)
+if(${LINUX})
 CHECK_INCLUDE_FILES("sched.h" HAVE_SCHED)
+endif(${LINUX})
+if(${FREBSD})
+CHECK_INCLUDE_FILES("sched.h" HAVE_SCHED_FREEBSD)
```

Comment:
On 30-6-2016 11:38, Kefu Chai wrote:
> In CMakeLists.txt
> <https://github.com/ceph/ceph/pull/10024#discussion_r69102479>:
> 
>>  CHECK_INCLUDE_FILES("sched.h" HAVE_SCHED)
>> +endif(${LINUX})
>> +if(${FREBSD})
>> +CHECK_INCLUDE_FILES("sched.h" HAVE_SCHED_FREEBSD)
> 
> do we reference |HAVE_SCHED_FREEBSD| anywhere?

I am in one of the CC files, because it tried to used the Linux
scheduler routines. Which are of course not available on FreeBSD.

But FreeBSD does have sched.h, so I need to fix that.
Other solution would be to exclude the HAVE_SCHED_FREEBSD and make the
code deal with the fact that HAVE_SCHED is not defined. And then later
on when the scheduler affinitity code for FreeBSD is done, then fix the
CMakefile

But I also see a typo. FREBSD. that ain't going to work.